modules/jdbc/src/main/java/org/testcontainers/jdbc/ContainerDatabaseDriver.java

```diff
@@ -40,12 +41,23 @@
     private Driver delegate;
     private static final Map<String, Set<Connection>> containerConnections = new HashMap<>();
     private static final Map<String, JdbcDatabaseContainer> jdbcUrlContainerCache = new HashMap<>();
+    private static final Map<String, Map<String, String>> jdbcAliasMap = new HashMap<String, Map<String,String>>();
+	
     private static final Set<String> initializedContainers = new HashSet<>();
     private static final String FILE_PATH_PREFIX = "file:";
+    private static final String JDBC_ALIAS_PREFIX = "jdbc.alias.";
 
     static {
         load();
     }
+    
+    /**
+     * Parse properties in default constructor
+     */
+    public ContainerDatabaseDriver() {
+    	super();
+    	mapJBDCAliasProperties(TestcontainersConfiguration.getInstance().getProperties());    	
```

Comment:
Why cached? You can get the property when you encounter an alias

Reply:
@bsideup we don't know the aliases without reading the configuration at instanciation.
IMHO it's the easiest way to implement it here, otherwise we won't be able to find candiate containers with the alias and have to see if the name is an alias afterwards...

Reply:
but you don't need it at instantiation time.

The algorithm should be something like:
1. get the type from the JDBC URL
2. try to get alias by it
3. if not found, fallback to the default one (the behaviour before this PR)